extra/html-extra/Cva.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extra\Html;
+
+/**
+ * Class Variant Authority (Cva) resolver.
+ *
+ * The Cva concept is used to render multiple variations of components, applying
+ * a set of conditions and recipes to dynamically compose CSS class strings.
+ *
+ * @see https://cva.style/docs
+ *
+ * @doc https://symfony.com/bundles/ux-twig-component/current/index.html
+ *
+ * @author Mathéo Daninos <matheo.daninos@gmail.com>
+ *
+ * @experimental
+ */
+final class Cva
+{
+    /**
+     * @var list<string|null>
+     */
+    private array $base;
+
+    /**
+     * @param string|list<string|null> $base The base classes to apply to the component
+     */
+    public function __construct(
+        string|array  $base = [],
+        /**
+         * The variants to apply based on recipes.
+         *
+         * Format: [variantCategory => [variantName => classes]]
+         *
+         * Example:
+         *      'colors' => [
+         *          'primary' => 'bleu-8000',
+         *          'danger' => 'red-800 text-bold',
+         *       ],
+         *      'size' => [...],
+         *
+         * @var array<string, array<string, string|list<string>>>
+         */
+        private array $variants = [],
+
+        /**
+         * The compound variants to apply based on recipes.
+         *
+         * Format: [variantsCategory => ['variantName', 'variantName'], class: classes]
+         *
+         * Example:
+         *   [
+         *      'colors' => ['primary'],
+         *      'size' => ['small'],
+         *      'class' => 'text-red-500',
+         *   ],
+         *   [
+         *      'size' => ['large'],
+         *      'class' => 'font-weight-500',
+         *   ]
+         *
+         * @var array<array<string, string|array<string>>>
+         */
+        private array $compoundVariants = [],
+
+        /**
+         * The default variants to apply if specific recipes aren't provided.
+         *
+         * Format: [variantCategory => variantName]
+         *
+         * Example:
+         *     'colors' => 'primary',
+         *
+         * @var array<string, string>
+         */
+        private array $defaultVariants = [],
+    )
+    {
+        $this->base = (array)$base;
+    }
+
+    public function apply(array $recipes, ?string ...$additionalClasses): string
+    {
+        $classes = [...$this->base];
```

Comment:
```suggestion
        $classes = $this->base;
```

PHP arrays are copy-on-write. There is no reason to force a copy if not needed.